test/linter/test-versions.ts

```diff
@@ -159,91 +141,103 @@ const checkVersions = (
   ) as BrowserName[];
 
   for (const browser of browsersToCheck) {
-    if (validBrowserVersions[browser]) {
-      const supportStatement: InternalSupportStatement | undefined =
-        supportData[browser];
+    const supportStatement: InternalSupportStatement | undefined =
+      supportData[browser];
 
-      if (!supportStatement) {
-        if (realValuesRequired[category].includes(browser)) {
-          logger.error(chalk`{red {bold ${browser}} must be defined}`);
-        }
+    if (!supportStatement) {
+      if (realValuesRequired[category].includes(browser)) {
+        logger.error(chalk`{red {bold ${browser}} must be defined}`);
+      }
+
+      continue;
+    }
 
+    for (const statement of Array.isArray(supportStatement)
+      ? supportStatement
+      : [supportStatement]) {
+      if (statement === 'mirror') {
+        // If the data is to be mirrored, make sure it is mirrorable
+        if (!browsers[browser].upstream) {
+          logger.error(
+            chalk`{bold ${browser}} is set to mirror, however {bold ${browser}} does not have an upstream browser.`,
+          );
+        }
         continue;
       }
 
-      for (const statement of Array.isArray(supportStatement)
-        ? supportStatement
-        : [supportStatement]) {
-        if (statement === 'mirror') {
-          // If the data is to be mirrored, make sure it is mirrorable
-          if (!browsers[browser].upstream) {
-            logger.error(
-              chalk`{bold ${browser}} is set to mirror, however {bold ${browser}} does not have an upstream browser.`,
-            );
-          }
+      for (const property of ['version_added', 'version_removed']) {
+        const version = statement[property];
+        if (property == 'version_removed' && version === undefined) {
+          // Undefined is allowed for version_removed
```

Comment:
Nitpicking a bit, the value `undefined` isn't allowed, but the property can be missing:

```suggestion
          // version_removed is optional.
```